nixos/modules/services/networking/networkmanager.nix

```diff
@@ -169,11 +151,11 @@ in
         '';
       };
 
-      extraConfig = mkOption {
-        type = types.lines;
-        default = "";
```

Comment:
I don't know if this is a de facto standard, but even if many modules are moved to `settings`, often they also retain `extraConfig`. Some new modules are still adding `extraConfig`.

The benefit of this is obvious: we cannot predict that we will always generate configs for users that meet their exact needs. Some exceptions may occur inadvertently, so a guaranteed option is always good.

Reply:
Changing to settings is the preferred key if every value can be defined using nix data structures and it's the "RFC42 style". Changing the name also doesn't really matter here since the option would break either way because the datatype was changed.

Reply:
it would be possible to say, have extraConfig that gets smacked on the very end of the file, which is what `nix.settings` did, in spite of having structured config as the primary method, to allow for migration. Personally I am fine either way, the only real argument to not breaking it immediately and hard is to give time to migrate (especially for any downstream modules that need to be cross compatible across NixOS versions), but I don't think this is terrifically critical.

Reply:
Agreed + parsing a ini and converting it to nix is fairly trivial for any usecase where the config is more then like 5 lines, I basically automated that in https://github.com/Janik-Haag/nm2nix/pull/6